### PR TITLE
update: make run_report a standalone function

### PR DESCRIPTION
**Pull request by alchalu**
Makes it possible to run report as a standalone function with only data as an input

### CODE REVIEW

**Comment by tinneuro** [comment]
Hi @alchalu ,

great work thanks for that! I tested it and everything is working for me. While testing I had two minor remarks:

- I think it would be nice if we change the order of arguments and let data = be first and then after that report_format. For me this is more intuitive. 
- New feature: It would be nice if we had an additional argument where the user could specify where the report is saved on his computer. Of course there would need to be some check whether this folder exists on the computer

As always could you also open up a new section in NEWS.md and write that you implemented the standalone report function?

Thanks!

**Comment by tinneuro** [comment]
Hi @alchalu ,

do you think you can incooperate the changes this week?

**Comment by alchalu** [comment]
Hi @tinneuro, 

thank you for your comments! 

- I changed the order of the inputs and add output_dir from rmarkdown::render to the inputs. Now the directory can be specified. 
- I also realized that I hadn't implemented the use of the intervention_date. That should be fixed now. 
- The changes are documented in the NEWS.md

File: R/run_report.R [hunk 1]
```diff
@@ -1,17 +1,22 @@
...
 #'
-#' @param report_format format of the report: HTML or DOCX
+#' If executed as a standalone function, all filtering must
+#' be performed beforehand.
+#' This function is also invoked within the app.
+#'
 #' @param data data.frame containing surveillance data in linelist format
-#' @param algo algorithm to be used
+#' @param report_format format of the report: HTML or DOCX
+#' @param method algorithm to be used
```
Comment (tinneuro) [inline R/run_report.R:9]: Can you maybe add the options the user has here, i.e. all the method names

**Comment by tinneuro** [comment]
When I try to run the report using the following settings:
run_report(input_example, method = "glm mean") I get an error message and it says that it quitted from lines 50-52 in the SignalDetectionReport.Rmd and Missing value, where TRUE/FALSE is needed. Same for the other glm methods. Can you investigate this?

**Comment by tinneuro** [comment]
It is great that the user now has the opportunity to specify an output_dir. Can you also add this to the NEWS that this is possible now?

**Comment by tinneuro** [comment]
Hi @alchalu,

can you manage to work on this this week? Thanks!

**Comment by alchalu** [comment]
Done. Sorry for the delay!

**Comment by tinneuro** [comment]
Hi :) 
Thanks for the updates, unfortunately I still run into a bug. 
When I try to execute run_report() just like this it does not work for me. Maybe it is some problem with my packages? Does it work for you? It says Quitting from lines 50-52 [unnamed-chunk-2] (SignalDetectionReport.Rmd)

**Comment by tinneuro** [comment]
I finally managed to get it running. I had a weird error that the png() device is not working and after some googling found out that this is a Windows problem when the Rproject lies in a folder with a long path this can happen. So I changed the location to a shorter path and it worked.
https://stackoverflow.com/questions/45113597/error-unable-to-start-png-device
I will finally review the rest tomorrow.

**Comment by tinneuro** [comment]
Hi :) I found one more issue. The run_report() function works well. But when I use the app and click on the "Create Report" Button the report is generated and also already saved automatically under a automatically generated name i.e. file3b744c796c38.docx but in the menu which then pops up where the user can actually select where to save the report "report-downloadReport.htm" this shows up instead of the report. Can we make it again that the correct .html or .docx shows up there and that it is named SignalDetectionReport and not saved beforehand with some automatic generated filename?

**Comment by tinneuro** [comment]
Furthermore when using the function I realised that the users of the app are used to the algorithm names they see in the app i.e. FarringtonFlexible instead of "farrington" we use in the background and so on. Thus I decided to change the code such that these method names we use in the app are also used in the run_report function. I will commit this now. Maybe you can review this? I will also add some tests in my next commit.

**Comment by alchalu** [comment]
Thank you so much for the changes and the test! They all look good to me.

**Comment by alchalu** [comment]
I just pushed a quick fix for the download directory. Now the default download directory is used when downloading from the app and the current working directory is used when running `run_report()` directly from an R session.

**Comment by alchalu** [comment]
I think the branch can finally be merged. @tinneuro can you do this? I can't merge my own request and Norbert is on vaccation. Thank you so  much!

**Comment by tinneuro** [comment]
Great thanks for the fix! It looks all good now.

**Review by tinneuro — APPROVED** [review]

**Pull request merged**